### PR TITLE
GraphQL Integration

### PR DESCRIPTION
**Pull request by Itay4**
## Status
- [x] Ready (still missing integration logo)

## Description
Generic GraphQL integration to interact with GraphQL server APIs

## Minimum version of Demisto
- [x] 5.0.0

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/GraphQL/Integrations/GraphQL/GraphQL.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import demistomock as demisto
+from CommonServerPython import *
+from CommonServerUserPython import *
+
+# Disable insecure warnings
+urllib3.disable_warnings()
+
+
+def execute_query(client: Client, args: Dict) -> CommandResults:
+    query = gql(args.get('query'))
```
Comment (anara123) [inline Packs/GraphQL/Integrations/GraphQL/GraphQL.py:16]: lets add an argument, `key_field`, if provided then you will pass it to `outputs_key_field`

**Review by anara123 — CHANGES_REQUESTED** [review]
add unit test

**Comment by Itay4** [comment]
> add unit test

i dont see the added value for unit test here, as there is no logic implemented and unit test would cover exactly what the test playbook covers

**Review by glicht — COMMENTED** [review]
Need to include description and update the image

File: Packs/GraphQL/Integrations/GraphQL/GraphQL.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+def main() -> None:
+    command = demisto.command()
+    try:
+        params = demisto.params()
+        request_params = {
+            'url': params.get('url'),
+            'verify': not params.get('insecure', False),
+            'retries': 3,
+        }
+        if params.get('credentials'):
```
Comment (glicht) [inline Packs/GraphQL/Integrations/GraphQL/GraphQL.py:36]: Let's also add support for specifying a header as we did in other integrations so a user can authenticate with a token. Such as in the github api: https://developer.github.com/v4/guides/forming-calls/#authenticating-with-graphql

File: Packs/GraphQL/Integrations/GraphQL/GraphQL.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                params.get('credentials').get('identifier'), params.get('credentials').get('password')
+            )
+        if params.get('headers'):
+            try:
+                request_params['headers'] = json.loads(params.get('headers'))
+            except json.JSONDecodeError as e:
+                raise ValueError(f'Headers are not in valid JSON format: {e}')
+
+        transport = RequestsHTTPTransport(**request_params)
+        if not params.get('proxy'):
```
Comment (glicht) [inline Packs/GraphQL/Integrations/GraphQL/GraphQL.py:47]: let's use `handle_proxy` from common. There is added stuff we do there such as dealing with `REQUESTS_CA_BUNDLE` env vars.

**Comment by ShirleyDenkberg** [comment]
@glicht @anara123 
For some reason, I am unable to update the README. I get an error when I press Comment.
The only change I want to make is to update the description of the outputs_key_field argument.  It should be the following:  **Primary key field in the response to unique the object in the context data.** 
Otherwise Doc review is done.

**Comment by anara123** [comment]
@ShirleyDenkberg seems you were able to update the readme and the yml

File: Packs/GraphQL/Integrations/GraphQL/GraphQL.py (hunk 2 above)
Comment (Itay4) [reply Packs/GraphQL/Integrations/GraphQL/GraphQL.py:36]: @glicht for that i added the `headers` parameter, dont you think its enough?

File: Packs/GraphQL/Integrations/GraphQL/GraphQL.yml [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+      required: false
+      secret: false
+    - default: false
+      description: 'A comma-separated list of values, for example: true,4,3.5'
+      isArray: true
+      name: variables_values
+      required: false
+      secret: false
+    - default: false
+      description: 'An optional comma-separated list of types, for example: Boolean,Int,Float.
```
Comment (glicht) [inline Packs/GraphQL/Integrations/GraphQL/GraphQL.yml:111]: I don't think there is a difference between int and float. It should be simply number and we should auto detect both int and float. Also let's make the types lower case: string,boolean,number.

Comment (Itay4) [reply Packs/GraphQL/Integrations/GraphQL/GraphQL.yml]: done

**Review by glicht — APPROVED** [review]

**Review by anara123 — APPROVED** [review]

**Pull request merged**